src/corehost/cli/deps_resolver.cpp

```diff
@@ -151,68 +340,21 @@ pal::string_t deps_resolver_t::resolve_coreclr_dir(
             trace::verbose(_X("Deps has no coreclr entry."));
         }
 
-        // Package cache.
-        pal::string_t coreclr_cache;
-        if (!package_cache_dir.empty())
-        {
-            if (deps->has_coreclr_entry())
-            {
-                const deps_entry_t& entry = deps->get_coreclr_entry();
-                trace::verbose(_X("Probing for CoreCLR package=[%s][%s] in package cache=[%s]"), entry.library_name.c_str(), entry.library_version.c_str(), package_cache_dir.c_str());
-                if (entry.to_hash_matched_path(package_cache_dir, &coreclr_cache))
-                {
-                    return get_directory(coreclr_cache);
-                }
-            }
-        }
-
-        // Deps directory: lookup relative path if portable, else look sxs.
-        if (is_portable)
-        {
-            pal::string_t coreclr_portable;
-            if (deps->has_coreclr_entry())
-            {
-                const deps_entry_t& entry = deps->get_coreclr_entry();
-                trace::verbose(_X("Probing for CoreCLR package=[%s][%s] in portable app dir=[%s]"), entry.library_name.c_str(), entry.library_version.c_str(), deps_dir.c_str());
-                if (entry.to_full_path(deps_dir, &coreclr_portable))
-                {
-                    return get_directory(coreclr_portable);
-                }
-            }
-        }
-        else
-        {
-            // App main dir or standalone app dir.
-            trace::verbose(_X("Probing for CoreCLR in deps directory=[%s]"), deps_dir.c_str());
-            if (coreclr_exists_in_dir(deps_dir))
-            {
-                return deps_dir;
-            }
-        }
-
-        // Packages dir.
-        pal::string_t coreclr_package;
-        if (!package_dir.empty())
+        // App/FX main dir or standalone app dir.
+        trace::verbose(_X("Probing for CoreCLR in deps directory=[%s]"), deps_dir.c_str());
+        if (coreclr_exists_in_dir(deps_dir))
         {
-            if (deps->has_coreclr_entry())
-            {
-                const deps_entry_t& entry = deps->get_coreclr_entry();
-                trace::verbose(_X("Probing for CoreCLR package=[%s][%s] in packages dir=[%s]"), entry.library_name.c_str(), entry.library_version.c_str(), package_dir.c_str());
-                if (entry.to_full_path(package_dir, &coreclr_package))
-                {
-                    return get_directory(coreclr_package);
-                }
-            }
+            return deps_dir;
         }
 
         return pal::string_t();
     };
 
-    trace::info(_X("--- Starting CoreCLR Proble from app deps.json"));
-    pal::string_t clr_dir = process_coreclr(m_portable, app_dir, m_deps.get());
+    trace::info(_X("--- Starting CoreCLR Probe from app deps.json"));
+    pal::string_t clr_dir = process_coreclr(m_portable, m_app_dir, m_deps.get());
     if (clr_dir.empty() && m_portable)
```

Comment:
If app deps does not have a CoreCLR entry AND app is portable should you check for CoreCLR in FX. Otherwise, you should fail.